lib/configSchema.js

```diff
@@ -0,0 +1,75 @@
+'use strict';
+
+const functionNamePattern = '^[a-zA-Z0-9-_]+$';
+
+const schema = {
+  type: 'object',
+  properties: {
+    service: {
+      type: 'object',
+      properties: {
+        name: { pattern: '^[a-zA-Z][0-9a-zA-Z-]+$' },
+        awsKmsKeyArn: { pattern: '^arn:(aws[a-zA-Z-]*)?:kms:[a-z0-9-]+-\\d+:\\d{12}:[^\\s]+$' },
+      },
+      additionalProperties: false,
+    },
+
+    custom: { type: 'object', properties: {}, required: [] },
+    plugins: { type: 'array', items: { type: 'string' } },
+    resources: { type: 'object' },
```

Comment:
Let's add a TODO comment (to be defined in separate PR), so we do not loose it, and do not assume that loose schema is intended in such case